src/components/PatronDetail/PatronDetail.js

```diff
@@ -1,55 +1,99 @@
 import React, { memo } from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
+import classNames from 'classnames';
 
 import {
   Row,
   Col,
   NoValue,
   FormattedDate,
+  KeyValue,
+  Loading,
 } from '@folio/stripes/components';
 import { ProfilePicture } from '@folio/stripes/smart-components';
 
-import { getFullName } from '../../util';
 import css from './PatronDetail.css';
+import { usePatronGroup } from '../../hooks';
 
 const PatronDetail = memo(({ user, isUserProfilePicConfigEnabledForTenant }) => {
+  const { data: patronGroup, isLoading: isPatronGroupLoading } = usePatronGroup(user?.patronGroup);
   const profilePictureLink = user?.personal?.profilePictureLink;
 
-  const getUserValue = () => {
-    return (
-      <Row className={css.marginTop}>
-        <Col xs={3}>
-          <strong>
-            {getFullName(user)}
-          </strong>
+  const renderBorrowerDetails = () => (
+    <div
+      className={
+        classNames(css.borrowerDetails, { [`${css.borrowerWhenProfilePicConfigActive}`]: isUserProfilePicConfigEnabledForTenant })
+      }
+    >
+      <Row>
+        <Col xs={2}>
+          <KeyValue
+            label={
+              <FormattedMessage
+                id={user?.personal?.preferredFirstName ? 'ui-reading-room.userDetail.preferredFirstName' : 'ui-reading-room.userDetail.firstName'}
+                tagName="strong"
+              />
+            }
+            value={user?.personal?.preferredFirstName || user?.personal?.firstName || <NoValue />}
```

Comment:
move `user?.personal?.preferredFirstName` to a variable as it is used in several places, please